pydantic/main.py

```diff
@@ -607,8 +601,11 @@ def validate_model(  # noqa: C901 (ignore complexity)
             if not model.__config__.validate_all and not field.validate_always:
                 values[name] = value
                 continue
-        elif check_extra:
-            names_used.add(field.name if using_name else field.alias)
+        else:
+            fields_set.add(name)
```

Comment:
Maybe `fields_set` and `names_used` are the same here, if so, this is a cleaner approach. If not I think we need a comment to explain the difference.

Reply:
`fields_set` is always the name of the field, `names_used` are the names contained in `input_data`, which may be aliases.

Reply:
ok, makes sense. Could you add a comment saying this to avoid confusion.